pkg/services/sqlstore/migrations/usermig/user_lowercase_login_and_email.go

```diff
@@ -0,0 +1,90 @@
+package usermig
+
+import (
+	"strings"
+
+	"github.com/grafana/grafana/pkg/services/sqlstore/migrator"
+	"github.com/grafana/grafana/pkg/services/user"
+	"xorm.io/xorm"
+)
+
+const (
+	LowerCaseUserLoginAndEmail = "update login and email fields to lowercase"
+)
+
+// Service accounts login were not unique per org. this migration is part of making it unique per org
+// to be able to create service accounts that are unique per org
+func AddLowerCaseUserLoginAndEmail(mg *migrator.Migrator) {
+	mg.AddMigration(LowerCaseUserLoginAndEmail, &UsersLowerCaseLoginAndEmail{})
+}
+
+var _ migrator.CodeMigration = new(UsersLowerCaseLoginAndEmail)
+
+type UsersLowerCaseLoginAndEmail struct {
+	migrator.MigrationBase
+}
+
+func (p *UsersLowerCaseLoginAndEmail) SQL(dialect migrator.Dialect) string {
+	return "code migration"
+}
+
+func (p *UsersLowerCaseLoginAndEmail) Exec(sess *xorm.Session, mg *migrator.Migrator) error {
+	// Get all users
+	users := make([]*user.User, 0)
+	err := sess.Table("user").Find(&users)
+	if err != nil {
+		return err
+	}
+
+	for _, usr := range users {
+		/*
+			LOGIN
+		*/
+		lowerLogin := strings.ToLower(usr.Login)
+
+		// Check if lower login exists
+		existingLowerCasedUserLogin := &user.User{}
+
+		// lowercaseexists in database
+		hasLowerCasedLogin, err := sess.Table("user").Where("login = ?", lowerLogin).Get(existingLowerCasedUserLogin)
+		if err != nil {
+			return err
+		}
```

Comment:
thought: you could avoid an extra DB query for each user if you created a map with all logins and emails (you've already fetched all the users, so you just need to loop over them and store the records in a map).

Reply:
this is true, i was thinking we would miss a case if we would do that. let me have a think and a look through.

I actually had a map of the logins before